grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -745,19 +748,19 @@ class StockMovementService {
         }
     }
 
-    def getStockMovementItems(String id, String stepNumber, String max, String offset) {
+    def getStockMovementItems(String id, Integer stepNumber, Integer max, Integer offset, Boolean createNewPicklist = true) {
```

Comment:
I don't remember why we these arguments are Strings, but I would be careful with unintended consequences if you haven't changed all String stepNumber variables to Integer throughout the code (lines 733-739). 

With that said, I'm fine with it as long as we make the change throughout or as long as there some sort of implicit coercion happening that allows for the case statements and comparisons to essentially treat the following as true.
```
assert 1 == "1"
assert "1".equals(1)
```

If not, we should revert.

If you are going to boy scout this one, you could go one step further and turn these into enums.
```
OutboundWorkflowState.valueOf(stepNumber) == OutboundWorkflowState.REVISE_ITEMS
```
and create an enum 
```
enum OutboundWorkflowState { 

    CREATE_HEADER(1),
    ADD_ITEMS(2),
    REVISE_ITEMS(3),
    PICK_ITEMS(4),
    PACK_ITEMS(5),
    SEND_SHIPMENT(6)

    Integer stepNumber

    OutboundWorkflowState(int stepNumber) {
        this.stepNumber = stepNumber
    }

}
```

Reply:
I am ok with this change, but please make sure you did not miss anything that is still passing the string